tools/src/main/scala/org/apache/spark/sql/rapids/tool/ui/QualificationReportGenerator.scala

```diff
@@ -0,0 +1,119 @@
+import java.text.SimpleDateFormat
+import java.util.Date
+
+import com.nvidia.spark.rapids.tool.qualification.Qualification
+import org.json4s.DefaultFormats
+import org.json4s.jackson.Serialization
+
+import org.apache.spark.SparkConf
+import org.apache.spark.internal.Logging
+import org.apache.spark.util.Utils.getPropertiesFromFile
+
+class QualificationReportGenerator(
+    conf: SparkConf,
+    provider: Qualification) {
+
+  import QualificationReportGenerator._
+
+  // TODO: use HDFS API to create the logoutput if necessary
+  val assetsFolder = new File(getAssetsPath)
+  val outputWorkDir = new File(provider.getReportOutputPath, "ui-report")
+  val destinationFolder = {
+    val timestamp = new SimpleDateFormat("yyyyMMddHHmm").format(new Date())
+    new File(s"${outputWorkDir}-${timestamp}")
+  }
+
+  def getAssetsPath(): String = s"${UI_HOME}${File.separator}${RAPIDS_UI_ASSETS_DIR}"
+
+  def copyAssetFolder(src: File, dest: File): Unit = {
+    if (src.isDirectory) {
+      if (!dest.exists()) {
+        dest.mkdirs()
+      }
+      src.listFiles().foreach { srcDir =>
+        copyAssetFolder(srcDir, new File(dest, srcDir.getName))
+      }
+    } else {
+      new FileOutputStream(dest) getChannel() transferFrom(
+        new FileInputStream(src) getChannel(), 0, Long.MaxValue)
+    }
+  }
+
+  def launch(): Unit = {
+    copyAssetFolder(assetsFolder, destinationFolder)
+    generateJSFiles
+  }
+
+  def generateJSFiles(): Unit = {
+    implicit val formats = DefaultFormats
+    val appInfoRecs = Serialization.write(provider.getListing())
+    val infoSummary = Serialization.write(provider.getAllApplicationsInfo())
+    val dataSourceInfo = Serialization.write(provider.getDataSourceInfo())
+    val qualInfoSummaryContent =
+      s"""
+         |
+         |let appInfoRecords =
+         |\t${appInfoRecs};
+         |let qualificationRecords =
+         |\t${infoSummary};
+         |let dataSourceInfoRecords =
+         |\t${dataSourceInfo};
+       """.stripMargin
+    val outputFile = new File(destinationFolder, s"js/mock-data.js")
+    val pw = new PrintWriter(outputFile)
+    pw.write(qualInfoSummaryContent)
+    pw.close
+  }
+}
+
+object QualificationReportGenerator extends Logging {
+  val RAPIDS_UI_PREFIX = "spark.rapids.tool.qualification.ui."
+  val UI_HOME = getClass.getResource("/ui").getPath
+  val RAPIDS_UI_CONF_DIR = "conf"
+  val RAPIDS_UI_ASSETS_DIR = "assets"
+  val RAPIDS_UI_CONF_FILE = "qual-ui-report-defaults.conf"
+
+  private val conf = new SparkConf
+
+  def createQualReportGenerator(
+      provider: Qualification,
+      propFilePath: String = null): Unit = {
+    loadDefaultSparkProperties(conf, propFilePath)
+    val generator = new QualificationReportGenerator(conf, provider)
+    generator.launch()
+  }
+
+  def loadDefaultSparkProperties(conf: SparkConf, filePath: String = null): String = {
+    val propPath = Option(filePath).getOrElse(getDefaultUIProperties())
```

Comment:
I tried running this locally on my machine and got this Exception while reading the default properties file. Looks like the path(which is in String) is incorrect. Please let me know if I am missing something.

Run command: `java -cp rapids-4-spark-tools_2.12-22.06.0-SNAPSHOT.jar:$SPARK_HOME/jars/*  com.nvidia.spark.rapids.tool.qualification.QualificationMain /home/nartal/spark-rapids/spark-rapids/tools/src/test/resources/spark-events-qualification/*`

Exception:
```
INFO ToolTextFileWriter: Summary report output location: rapids_4_spark_qualification_output/rapids_4_spark_qualification_output.log
Exception in thread "main" java.lang.IllegalArgumentException: requirement failed: Properties file file:/home/nartal/spark-rapids/spark-rapids/tools/target/spark311/rapids-4-spark-tools_2.12-22.06.0-SNAPSHOT.jar!/ui/conf/qual-ui-report-defaults.conf does not exist
	at scala.Predef$.require(Predef.scala:281)
	at org.apache.spark.util.Utils$.getPropertiesFromFile(Utils.scala:2145)
	at org.apache.spark.sql.rapids.tool.ui.QualificationReportGenerator$.loadDefaultSparkProperties(QualificationReportGenerator.scala:110)
	at org.apache.spark.sql.rapids.tool.ui.QualificationReportGenerator$.createQualReportGenerator(QualificationReportGenerator.scala:103)
	at com.nvidia.spark.rapids.tool.qualification.Qualification.launchUIReportGenerator(Qualification.scala:155)
	at com.nvidia.spark.rapids.tool.qualification.Qualification.qualifyApps(Qualification.scala:89)
	at com.nvidia.spark.rapids.tool.qualification.QualificationMain$.mainInternal(QualificationMain.scala:92)
	at com.nvidia.spark.rapids.tool.qualification.QualificationMain$.main(QualificationMain.scala:33)
	at com.nvidia.spark.rapids.tool.qualification.QualificationMain.main(QualificationMain.scala)
```

Reply:
My bad! I was running the command line adding the classes to the cp.
I will work on fixing the resource loading.
Meanwhile, the `cp` needs to include the classes folder.

Reply:
This comment is not valid anymore since `qual-ui-report-defaults.conf` is removed in the latest commit.